### DISCUSSION TITLE
Why is _input_reader_linux.py a polling thread? (and not async?)

### DISCUSSION THREAD

**Discussion by jfasch** [Textualize/textual#4363] [Q&A]
I've just finished my first app, and looked at strace output for another reason. I see that the process has two threads, with one of them polling on stdin in a 0.1s interval. Apparently, the polling is done so that the thread notices a quit-request within at most 0.1s.
Why is this? Can't terminal input be taken from within asyncio's event loop , without polling?

Thanks,
Joerg

**Comment by willmcgugan** [comment]
Congrats on your first app.

asyncio doesn't offer an way of reading from file descriptors, only sockets.

Only the exit event is polled 10 times a second, which takes negligible CPU. The input uses a selector, so that it can respond to input as it arrives.
